src/runtime_src/core/pcie/emulation/common_em/config.h

```diff
@@ -89,6 +90,89 @@ namespace xclemulation{
   const int VIVADO_MIN_VERSION = 2000;
   const int VIVADO_MAX_VERSION = 2100;
 
+//sparse log utility
+enum class eEmulationType{
+  eSw_Emu,
+  eHw_Emu
+};
+
+struct sParseLog
+{
+  std::ifstream mFileStream;
+  std::string mFileName;
+  std::atomic_bool mFileExists;
+  std::vector<std::string> mMatchedStrings;
+  eEmulationType mEmuType;
+
+  sParseLog(const std::string& iDeviceLog, eEmulationType iType, const std::vector<std::string>& iMatchedStrings)
+      : mFileName(iDeviceLog)
+      , mFileExists{false}
+      , mMatchedStrings(iMatchedStrings)
+      , mEmuType(iType)
+  {
+
+  }
+
+  void check_simulator_status()
+  {
+    std::string line;
+    while (std::getline(mFileStream, line))
+    {
+      for (const auto &StringData : mMatchedStrings)
+      {
+        if (line.find(StringData) != std::string::npos)
+        {
+          if (eEmulationType::eSw_Emu == mEmuType)
+            print_user_msg();
+          else if (eEmulationType::eHw_Emu == mEmuType)
+          {
+            if (!StringData.compare("Exiting xsim") || !StringData.compare("FATAL_ERROR"))
+              print_user_msg();
+            else
+              std::cout << line << '\n';
+          }
+        }
+      }
+    }
+  }
+
+  void print_user_msg()
+  {
+    switch (mEmuType) {
+      case eEmulationType::eSw_Emu:
+        std::cout << "Received request to end the application. Press Cntrl+C to exit the application." << std::endl;
```

Comment:
```suggestion
        std::cout << "Received request to end the application. Press Control+C to exit the application." << std::endl;
```
Why not a `std::exit()` or what ever?

Reply:
Please add some design comments, as discussed off-line.